src/test/java/uk/nhs/digital/nhsconnect/lab/results/utils/TimestampServiceTest.java

```diff
@@ -0,0 +1,24 @@
+package uk.nhs.digital.nhsconnect.lab.results.utils;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class TimestampServiceTest {
+    @Test
+    public void whenGettingTimestamp_thenPrecisionIsMilliseconds() {
+        var instant = new TimestampService().getCurrentTimestamp();
+        long remainder = instant.getNano() % 1000000; // nanoseconds per millisecond
```

Comment:
Consider using underscores to make the big number understandable at a glance
```suggestion
        long remainder = instant.getNano() % 1_000_000; // nanoseconds per millisecond
```